docs/data/material/guides/custom-component/custom-component.md

```diff
@@ -0,0 +1,335 @@
+
+- const Stat = React.forwardRef(function Stat(props, ref) {
++ const Stat = React.forwardRef(function Stat(inProps, ref) {
++   const props = useThemeProps({ props: inProps, name: 'MuiStat' });
+    const { value, unit, ...other } = props;
+
+    return (
+      <StatRoot ref={ref} {...other}>
+        <StatValue>{value}</StatValue>
+        <StatUnit>{unit}</StatUnit>
+      </StatRoot>
+    );
+  });
+```
+
+Then your users can customize the default props of your component like this:
+
+```js
+import { createTheme } from '@mui/material/styles';
+
+const theme = createTheme({
+  components: {
+    MuiStat: {
+      defaultProps: {
+        variant: 'outlined',
+      },
+    },
+  },
+});
+```
+
+## TypeScript
+
+If you use TypeScript, you should create interfaces for the component props and ownerState.
+
+```js
+interface StatProps {
+  value: number | string;
+  unit: string;
+  variant?: 'outlined';
+}
+
+interface StatOwnerState extends StatProps {
+  // …key value pairs for the internal state that you want to style the slot
+  // but don't want to expose to the users
+}
+```
+
+Then you can use them in the component and slots.
+
+```js
+const StatRoot = styled('div', {
+  name: 'MuiStat',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})<{ ownerState: StatOwnerState }>(({ theme, ownerState }) => ({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: theme.spacing(0.5),
+  padding: theme.spacing(3, 4),
+  backgroundColor: theme.palette.background.paper,
+  borderRadius: theme.shape.borderRadius,
+  boxShadow: theme.shadows[2],
+  letterSpacing: '-0.025em',
+  fontWeight: 600,
+  // typed-safe access to the `variant` prop
+  ...(ownerState.variant === 'outlined' && {
+    border: `2px solid ${theme.palette.divider}`,
+    boxShadow: 'none',
+  }),
+}));
+
+// …do the same for other slots
+
+const Stat = React.forwardRef<HTMLDivElement, StatProps>(function Stat(inProps, ref) {
+  const props = useThemeProps({ props: inProps, name: 'MuiStat' });
+  const { value, unit, variant, ...other } = props;
+
+  const ownerState = { ...props, variant };
+
+  return (
+    <StatRoot ref={ref} ownerState={ownerState} {...other}>
+      <StatValue ownerState={ownerState}>{value}</StatValue>
+      <StatUnit ownerState={ownerState}>{unit}</StatUnit>
+    </StatRoot>
+  );
+});
+```
+
+Finally, you have to add the Stat component the theme types.
+
+```ts
+import {
+  ComponentsProps,
+  ComponentsOverrides,
+  ComponentsVariants,
+  Theme as MuiTheme,
+} from '@mui/material/styles';
+// TODO: change the import path
+import { StatProps } from '/path/to/Stat';
+
+type Theme = Omit<MuiTheme, 'components'>;
+
+// shut off automatic exporting for the `Theme` above
+export {};
+
+interface CustomComponentsPropsList {
+  // TODO: change to your component name
+  MuiStat?: StatProps;
+}
+
+declare module '@mui/material/styles' {
+  interface ComponentsPropsList extends LabComponentsPropsList {}
```

Comment:
```suggestion
  interface ComponentsPropsList extends LabComponentsPropsList {}
```

I wouldn't use the lab here, it would be confusing for people not familiar with it. I would propose we just add a note that if you are using the `@mui/lab` you should extend this interface.